src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -652,6 +660,10 @@ private function addService($id, $definition)
         if ($definition->isSynthetic()) {
             $code .= sprintf("        throw new RuntimeException('You have requested a synthetic service (\"%s\"). The DIC does not know how to construct this service.');\n    }\n", $id);
         } else {
+            if ($definition->isDeprecated()) {
+                $code .= sprintf("        @trigger_error('The service %s has been marked as deprecated. You should stop using it.', E_USER_DEPRECATED);\n\n", $id);
```

Comment:
`suggestion` wouldn't be enough: we also usually put **e.g.** the Symfony version where something has been deprecated. Having a message is the most generic.
Why *too soon*? Is it a big change? I don't think so but I may miss something.